index/scorch/optimize_knn.go

```diff
@@ -85,8 +85,16 @@ func (o *OptimizeVR) Finish() error {
 					wg.Done()
 				}()
 				for field, vrs := range o.vrs {
-					vecIndex, err := segment.InterpretVectorIndex(field,
-						o.requiresFiltering, origSeg.deleted)
+					var vecIndex segment_api.VectorIndex
+					var err error
+					// Noop if the vector field or its index data is supposed to be deleted
+					if info, ok := o.snapshot.updatedFields[field]; ok && (info.Deleted || info.Index) {
+						vecIndex, err = segment.InterpretVectorIndex("",
```

Comment:
we can just short circuit the downstream operations by returning here right? like we know that an empty string field name is never going to exist in the index files and hence a backing faiss index won't be formed. so let's not perform calls to zap and keep it simple like

```
if info, ok := o.snapshot.updatedFields[field]; ok && (info.Deleted || info.Index) {
     return saying field doesn't exist
}
vecIndex, err := segment.InterpretVectorIndex(field,
		o.requiresFiltering, origSeg.deleted)
```

so we can just do a no-op or return an error saying "field doesn't exist"